### PR TITLE
add requirement, fix notice, fix !array value

### PR DESCRIPTION
**Pull request by bscheshirwork**

### CODE REVIEW

File: src/Connection.php [hunk 1]
```diff
@@ -253,7 +253,7 @@ public function makeRequest($method, $url, $options = [], $body = null, $raw = f
      */
     private function createUrl($path, array $options = [])
     {
-        $options = array_merge($this->getAuth(), $options);
+        $options = array_merge($this->getAuth() ?: [], $options);
```
Comment (SilverFire) [inline src/Connection.php:256]: I think it's better to change the default value of `$_auth` to `[]`

File: src/Query.php [hunk 2]
```diff
@@ -24,10 +24,6 @@ class Query extends Component implements QueryInterface
 
     public $type;
 
-    public $where;
```
Comment (SilverFire) [inline src/Query.php:27]: Why did you remove these properties?

Comment (SilverFire) [inline src/Query.php:27]: They are used in [QueryBuilder::build()](https://github.com/hiqdev/yii2-hiart/blob/master/src/QueryBuilder.php#L48-L51)

File: src/Connection.php (hunk 1 above)
Comment (bscheshirwork) [inline src/Connection.php:256]: result of closure must be array too

File: src/Query.php (hunk 2 above)
Comment (bscheshirwork) [inline src/Query.php:27]: duplicate yii\db\QueryTrait property $where

**Comment by SilverFire** [comment]
Merged with minor adjustments 55d33d97d8490af4e47eeb32bbdb3d3e8c0af59b

**Comment by SilverFire** [comment]
Thank you!
